lxd/storage/drivers/driver_pure_util.go

```diff
@@ -439,9 +463,33 @@ func (p *pureClient) getHosts() ([]pureHost, error) {
 	return resp.Items, nil
 }
 
-// createHost creates a new host that can be associated with specific volumes.
-func (p *pureClient) createHost(hostName string) error {
-	req, err := p.createBodyReader(map[string]any{})
+// getHostByIQN retrieves an existing host that is configured with the given IQN.
+func (p *pureClient) getHostByIQN(iqn string) (*pureHost, error) {
+	hosts, err := p.getHosts()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, host := range hosts {
+		if slices.Contains(host.IQNs, iqn) {
+			return &host, nil
+		}
+	}
+
+	return nil, api.StatusErrorf(http.StatusNotFound, "Host with IQN %q not found", iqn)
+}
+
+// createHost creates a new host with provided initiator qualified names that can be associated
+// with specific volumes.
+func (p *pureClient) createHost(hostName string, qns []string) error {
+	body := make(map[string]any, 1)
+
+	switch p.driver.config["pure.mode"] {
+	case pureModeISCSI:
+		body["iqns"] = qns
```

Comment:
should we have a default case here that returns an error if the mode is unhandled?